core/src/test/scala/integration/kafka/server/MultipleListenersWithAdditionalJaasContextTest.scala

```diff
@@ -0,0 +1,46 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one
+  * or more contributor license agreements.  See the NOTICE file
+  * distributed with this work for additional information
+  * regarding copyright ownership.  The ASF licenses this file
+  * to you under the Apache License, Version 2.0 (the
+  * "License"); you may not use this file except in compliance
+  * with the License.  You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package integration.kafka.server
+
+import java.util.Properties
+
+import kafka.api.CustomKafkaServerSasl
+import kafka.server.MultipleListenersWithSameSecurityProtocolBaseTest
+import org.apache.kafka.common.network.ListenerName
+
+
+class MultipleListenersWithAdditionalJaasContextTest extends MultipleListenersWithSameSecurityProtocolBaseTest{
+
+  override def setSaslProperties(listenerName: ListenerName): Option[Properties] = {
+
+    val gssapiSaslProperties = kafkaClientSaslProperties("GSSAPI", dynamicJaasConfig = true)
+    val plainSaslProperties = kafkaClientSaslProperties("PLAIN", dynamicJaasConfig = true)
+
+    if (listenerName.value == "SECURE_INTERNAL")
+      Some(plainSaslProperties)
+    else if (listenerName.value == "SECURE_EXTERNAL")
```

Comment:
It would be nice to have constants for these and any other strings that are shared between classes. You can also use pattern matching:

```scala
listenerName.value match {
  case SecureInternal => Some(...)
  case SecureExternal =>Some(...)
  case _ => None
```